packages/xo-web/src/xo-app/dashboard/health/index.js

```diff
@@ -538,6 +562,19 @@ export default class Health extends Component {
 
   _getSrUrl = sr => `srs/${sr.id}`
 
+  _getDuplicatedMacAddresses = createSelector(
+    () => this.props.vifsByMac,
+    vifsByMac => {
+      const duplicatedMacAddresses = []
+      for (const [macAddress, vifs] of Object.entries(vifsByMac)) {
+        if (vifs.length > 1) {
+          duplicatedMacAddresses.push(macAddress)
+        }
```

Comment:
What would be the reason for `vifs` to be empty?

Reply:
`vifs` can't be empty but can be equal to one.
If `vifs` are equal to one, then the mac address aren't duplicated

Reply:
Right!

You can call `.filter()` directly on the previous selector, though.

Reply:
The problem is if i use `.filter()` on the previous selector, this return me a new array with new key (0,1,2,..) instead of the `mac address` as key

Reply:
Calling `.filter()` on a selector that resolves with an object should give you a new selector that also resolves with an object. Is that what you meant?

Reply:
i mean, using `.filter()` on one selector, is replacing my `groupBy` collection with a new array of `VIFs` no ?